src/cripts/README.md

```diff
@@ -0,0 +1,453 @@
+# 1. Cripts
+
+Cripts, C-Scripts, is a set of wrappers and include files, for making simple ATS
+plugins easy to generate, modify or create from scratch. A key design here is that
+the Code is the Configuration, i.e. the intent really is to have a custom Cript file
+for every remap rule in the running system.
+
+ToDo: This document is not 100% updated with all features, but is at least a starting point.
+
+- [1. Cripts](#1-cripts)
+- [2. Building Cripts](#2-building-cripts)
+  - [2.1. Building a Cript](#21-building-a-cript)
+  - [2.2. Lint: Validating a Cript](#22-lint-validating-a-cript)
+  - [2.3. Clang-tidy](#23-clang-tidy)
+- [3. Writing Cripts](#3-writing-cripts)
+  - [3.1. Data types](#31-data-types)
+  - [3.2. Hooks](#32-hooks)
+  - [3.3. Processing and modifying headers](#33-processing-and-modifying-headers)
+  - [3.4. Processing and modifying URLs](#34-processing-and-modifying-urls)
+    - [3.4.1. Special case: Cache Key URL](#341-special-case-cache-key-url)
+  - [3.5. Accessing and modifying connections](#35-accessing-and-modifying-connections)
+  - [3.6. Overridable configurations](#36-overridable-configurations)
+  - [3.7. Pattern and identity matching](#37-pattern-and-identity-matching)
+  - [3.8. Cryptography functions](#38-cryptography-functions)
+  - [3.9. Various other utilities](#39-various-other-utilities)
+  - [3.10. Transaction contexts](#310-transaction-contexts)
+    - [3.10.1. Instance data (parameters)](#3101-instance-data-parameters)
+    - [3.10.2. Transaction data](#3102-transaction-data)
+- [4. Plugins Cript can mimic or replace](#4-plugins-cript-can-mimic-or-replace)
+  - [4.1. conf\_remap](#41-conf_remap)
+  - [4.2. cachekey](#42-cachekey)
+  - [4.3. header\_rewrite](#43-header_rewrite)
+  - [4.4. regex\_remap](#44-regex_remap)
+  - [4.5. geoip\_acl and maxmind\_acl](#45-geoip_acl-and-maxmind_acl)
+  - [4.6. tcpinfo](#46-tcpinfo)
```

Comment:
Is this an autogenerated table of contents? If so, to keep it updated, lets add `<!---` describing how to generate it.

I do this in the yahoo/proxy-verifier README.md:
https://github.com/yahoo/proxy-verifier/blob/e0f644464018198e5e0c248dbe40918eacb55ebb/README.md?plain=1#L1-L12